### DISCUSSION TITLE
Add a sql comment to generated queries

### DISCUSSION THREAD

**Discussion by kentlarsson** [sequelize/sequelize#15382] [Help & Questions] [answered]
Hi! Thanks for a great ORM!

I'm looking at [sqlcommenter](https://google.github.io/sqlcommenter/node/sequelize/) for adding some metadata through sql comments which can then be interpreted by Google [Query Insights](https://cloud.google.com/sql/docs/postgres/using-query-insights).

I've looked at the long list of hooks, but didn't really understand which ones could help me, if any of them can.

Ideally I would like to:

- Use a hook which can take a new extra argument "meta" (or something) which I then pass into all the standard methods like [findOne](https://sequelize.org/api/v6/class/src/model.js~model#static-method-findOne) etc when calling them.
- Then I could use the "meta" argument in my hook and add a sql comment based on it.
- I would include [cloud function name](https://cloud.google.com/functions) and JS function name where findOne is called that way, by passing it as a parameter. It would give me pretty great insights.

Is there anything like that? Or any nice way to tag on an sql comment for all queries at least? (for the latter I wouldn't get JS function name but still get the cloud function name which would improve things for us a lot)

(There is a wrapper for sequelize in the sqlcommenter repo, but I'd like to know if it's possible to do without wrapping sequelize and also because the wrapper doesn't fit my needs)

**Comment by ephys** [comment, marked as answer]
Hi :)

I'm not familiar with how Query Insights works, however:

- There is an issue to be able to pass custom options to methods, to be able to use them in hooks https://github.com/sequelize/sequelize/issues/14632
- I was going to say that you can modify the query before its execution using the `beforeQuery` hook, but it looks like that's not true: https://github.com/sequelize/sequelize/blob/main/src/sequelize.js#L709-L712. The SQL is made available after `beforeQuery`. If this is what you need, don't hesitate to open a feature request. Looks like sqlcommenter worked around it [by wrapping our `Query.run` method](https://github.com/google/sqlcommenter/blob/master/nodejs/sqlcommenter-nodejs/packages/sqlcommenter-sequelize/index.js#L46) 😅

**Reply by kentlarsson** [reply]
Hey!

It seems like [sqlcommenter](https://google.github.io/sqlcommenter/spec/) is some kind of open standard to basically use (abuse? :) ) sql comments to send key-value-pairs that then can be used by different tooling.

Thanks for digging into this, I really appreciate it!

I'll open a feature request then. I might wrap Query.run as well, but not being able to add some parameter to the actual query that's amongst the key-value-pairs makes it hard to send along the function name (i.e getUsers.name ) to get a better understanding of the query origin.

I'll mark this as the answer.

Have a nice weekend!

**Reply by ephys** [reply]
You should already be able to use `customOptions`. It's not officially supported yet but it works with most methods

I've opened a feature request for the ability to modify the SQL in `beforeQuery`: https://github.com/sequelize/sequelize/issues/15416

> Have a nice weekend!

Thanks! you too :)
